shared/src/main/scala/org/adridadou/openlaw/parser/template/printers/XHtmlAgreementPrinter.scala

```diff
@@ -154,18 +153,35 @@ case class XHtmlAgreementPrinter(preview: Boolean, paragraphEdits: ParagraphEdit
         case section@SectionElement(_, level, _, _, _, _) =>
           // partition out all content that will be within the newly defined sections
           val higherLevel = level - 1
-          val (content, remaining) = partitionAt(xs) { case SectionElement(_, thisLevel, _, _, _, _) if thisLevel === higherLevel => true }
-
+          val (content, remaining) = partitionAt(xs) { 
+            case SectionElement(_, thisLevel, _, _, _, _) if thisLevel === higherLevel => true 
+          }
           // Partition the elements into sections at this level
           val sections = partitionSections(level, section +: content)
-
-          val frag = ul(`class` := s"list-lvl-$level")(
-            sections.map { section =>
-              recurse(section._2, conditionalBlockDepth, true, { elems => Seq(li(elems)) })
-            }
-          )
-          tailRecurse(remaining, conditionalBlockDepth, inSection, { elems => continue(frag +: elems) } )
-
+          var afterBreak:Seq[AgreementElement] = Seq()
+          sections.map { section => 
+            // bifurcate the sections around the section break 
+            val (beforeBreak, after) = section._2.span(_ != Paragraph(List(FreeText(SectionBreak))))
+            afterBreak = afterBreak ++ after
+          }
+          if (afterBreak.size > 0) {
+            val frag = ul(`class` := s"list-lvl-$level")(
+              sections.map { section =>
```

Comment:
Unrelated to this fix, but since we're in here anyway it might increase readability to change the other tuple accessors here to use a locally bound variable instead, e.g.

```
sections.map { case (_, elements) =>
// section break elements and after should not be given the `ul/list` class
val filtered = elements filterNot afterBreak.contains
```